tests/lean/issue3232.lean.expected.out

```diff
@@ -0,0 +1,6 @@
+issue3232.lean:5:2-5:9: error: tactic 'apply' failed, failed to unify
+  @foo 42
+with
+  @foo 23
```

Comment:
Was
```
tactic 'apply' failed, failed to unify
  foo
with
  foo
```
before

Reply:
Could you add tests to show what happens when the differences are hidden by notation?

Reply:
I don't know much about notation, but happy to add any tests that you paste here :-)

(Even if it doesn’t work then yet it’s probably worth merging this as an incremental improvement.)

Reply:
Maybe this?
```
example : (1 : Nat) = 1 := by
  apply (rfl : (1 : Int) = 1)
```

Reply:
Before:
```
tactic 'apply' failed, failed to unify
  1 = 1
with
  1 = 1
```
now
```
tactic 'apply' failed, failed to unify
  @OfNat.ofNat Int 1 instOfNat = 1
with
  @OfNat.ofNat Nat 1 (instOfNatNat 1) = 1
```